runtime/src/test/java/org/corfudb/runtime/ClientHandshakeHandlerTest.java

```diff
@@ -0,0 +1,190 @@
+package org.corfudb.runtime;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.read.ListAppender;
+import io.netty.channel.embedded.EmbeddedChannel;
+import org.corfudb.protocols.service.CorfuProtocolMessage.ClusterIdCheck;
+import org.corfudb.protocols.service.CorfuProtocolMessage.EpochCheck;
+import org.corfudb.protocols.wireprotocol.ClientHandshakeHandler;
+import org.corfudb.runtime.proto.service.CorfuMessage.HeaderMsg;
+import org.corfudb.runtime.proto.service.CorfuMessage.PriorityLevel;
+import org.corfudb.runtime.proto.service.CorfuMessage.ProtocolVersionMsg;
+import org.corfudb.runtime.proto.service.CorfuMessage.RequestMsg;
+import org.corfudb.runtime.proto.service.CorfuMessage.ResponseMsg;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.corfudb.protocols.CorfuProtocolCommon.DEFAULT_UUID;
+import static org.corfudb.protocols.CorfuProtocolCommon.getUuidMsg;
+import static org.corfudb.protocols.service.CorfuProtocolBase.getHandshakeResponseMsg;
+import static org.corfudb.protocols.service.CorfuProtocolBase.getPingResponseMsg;
+import static org.corfudb.protocols.service.CorfuProtocolMessage.getHeaderMsg;
+import static org.corfudb.protocols.service.CorfuProtocolMessage.getResponseMsg;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class ClientHandshakeHandlerTest {
+
+    // Netty EmbeddedChannel for unit testing.
+    private EmbeddedChannel embeddedChannel;
+
+    // Testing appender for collecting log messages during test execution.
+    private ListAppender<ILoggingEvent> listAppender;
+
+    // Some utility variables needed for following test cases.
+    private final AtomicInteger requestCounter = new AtomicInteger();
+
+    private static final int DEFAULT_HANDSHAKE_TIMEOUT = 10;
+
+    private static final long FAKE_SERVER_VERSION = 111111111;
+
+    private static final UUID SERVER_NODEID = UUID.randomUUID();
+
+    /**
+     * A helper method that creates a basic message header populated
+     * with default values.
+     *
+     * @param ignoreClusterId   indicates if the message is clusterId aware
+     * @param ignoreEpoch       indicates if the message is epoch aware
+     * @return                  the corresponding HeaderMsg
+     */
+    private HeaderMsg getBasicHeader(ClusterIdCheck ignoreClusterId, EpochCheck ignoreEpoch) {
+        return getHeaderMsg(requestCounter.incrementAndGet(), PriorityLevel.NORMAL, 0L,
+                getUuidMsg(DEFAULT_UUID), getUuidMsg(DEFAULT_UUID), ignoreClusterId, ignoreEpoch);
+    }
```

Comment:
We should consider factoring/restructuring common RPC unit test code in a subsequent PR.

Reply:
Will do this in a following cleanup/refactor PR.